nixos/modules/services/cluster/kubernetes/kubelet.nix

```diff
@@ -308,56 +294,6 @@ in
           '';
           WorkingDirectory = top.dataDir;
         };
-        unitConfig.ConditionPathExists = kubeletPaths;
-      };
-
-      systemd.paths.kubelet = {
```

Comment:
Note that this didn't do what we probably wanted in the first place.  

This `systemctl start kubelet.service` any time either `PathExists=` or `PathChanged=` is matched.  Think of `.path` unit like `.timer` units.  A `.timer` unit isn't active for the lifetime for when the Calendar clause is active, it is _always_ active and gets pulled in by `timers.target` It merely activates another unit when the `Calendar` clause is matched.  Similarily,  a `.path` unit is part of `paths.target` and is _always active_ and then Starts a `.service` whenever some conditions changes.  It's a mistake I have made a lot too when first looking at `.path` units. 

Another example is a `.socket` unit. a `.socket` unit is always active (gets pulled in by `sockets.target`) but starts a `.service` whenever activity on a socket occurs.

Reply:
I hadn't even noticed that. Definitely something to keep in mind. That said, I'm not sure we *can* come up with an ordering scheme that's reasonably simple and still works in disparate environments, with all the combinations of daemons we can think of. We'll see. :)